src/iceberg/json_internal.h

```diff
@@ -74,44 +76,32 @@ Result<std::unique_ptr<SortOrder>> SortOrderFromJson(const nlohmann::json& json)
 ///
 /// \param[in] schema The Iceberg schema to convert.
 /// \return The JSON representation of the schema.
-nlohmann::json SchemaToJson(const Schema& schema);
-
-/// \brief Convert an Iceberg Type to JSON.
-///
-/// \param[in] type The Iceberg type to convert.
-/// \return The JSON representation of the type.
-nlohmann::json TypeToJson(const Type& type);
-
-/// \brief Convert an Iceberg SchemaField to JSON.
-///
-/// \param[in] field The Iceberg field to convert.
-/// \return The JSON representation of the field.
-nlohmann::json FieldToJson(const SchemaField& field);
-
-/// \brief Serializes a `SnapshotRef` object to JSON.
-///
-/// \param[in] snapshot_ref The `SnapshotRef` object to be serialized.
-/// \return A JSON object representing the `SnapshotRef`.
-nlohmann::json ToJson(const SnapshotRef& snapshot_ref);
-
-/// \brief Serializes a `Snapshot` object to JSON.
-///
-/// \param[in] snapshot The `Snapshot` object to be serialized.
-/// \return A JSON object representing the `snapshot`.
-nlohmann::json ToJson(const Snapshot& snapshot);
+nlohmann::json ToJson(const Schema& schema);
 
 /// \brief Convert JSON to an Iceberg Schema.
 ///
 /// \param[in] json The JSON representation of the schema.
 /// \return The Iceberg schema or an error if the conversion fails.
 Result<std::unique_ptr<Schema>> SchemaFromJson(const nlohmann::json& json);
 
+/// \brief Convert an Iceberg Type to JSON.
```

Comment:
nit: should we stick to the same doc pattern, `Serializes a 'xxx' object to JSON`.

Reply:
Sure, let me do this in a follow-up PR. This one gets large enough.